tests/perf/mt_ubench_spdlog.cpp

```diff
@@ -1,31 +1,67 @@
+/**
+ * *****************************************************************************
+ * \file perf/mt_ubench_spdlog.cpp
+ * \author Greg Law
+ * \brief L3: spdlog-multi-threaded logging micro perf-benchmarking
+ * \version 0.1
+ * \date 2024-06-25
+ *
+ * \copyright Copyright (c) 2024
+ *
+ * Usage: program-name [ <number-of-threads> ]
+ * *****************************************************************************
+ */
+#include <stdlib.h>
+#include <pthread.h>
 #include <spdlog/spdlog.h>
 #include <spdlog/sinks/basic_file_sink.h>
-#include <stdlib.h>
 
-#include <pthread.h>
+
+// Micro perf benchmarking workload parameter defaults
+constexpr int L3_PERF_UBM_NTHREADS = 10;
+constexpr int L3_PERF_UBM_NMSGS = (1000 * 1000);
+constexpr int L3_PERF_UBM_LOG_SIZE = (16 * 1024);
+
+#if __APPLE__
+#define L3_GET_TID()    pthread_mach_thread_np(pthread_self())
+#else
+#define L3_GET_TID()    syscall(SYS_gettid)
+#endif  // __APPLE__
 
 std::shared_ptr<spdlog::logger> logger;
 
+#if !defined(_POSIX_BARRIERS) || _POSIX_BARRIERS > 0
```

Comment:
This change is really needed to get this file compiling on my Mac/OSX. 

Strangely, the prev PR #91 clears CI-build jobs on Mac/OSX, so the pthread-library headers available on CI Mac machines have the right dependencies.

This "fix" seems needed only for my personal Mac. I don't have time to resolve the s/w components dependency on my machine, so would like to get this in for now. Sometime in the near future, I can investigate my package env later and undo this fix, if that's ok by you.